docs/support/tutorials/snakemake-puhti.md

```diff
@@ -0,0 +1,298 @@
+
+## Running Snakemake workflow on Puhti
+
+Snakemake workflow is one of the popular scientific workflows in the bioinformatics community although the workflow manager itself can be used in other scientific disciplines as well. The Snakemake enables scalable and reproducible scientific pipelines by chaining a series of rules in a fully-specified software environment. Snakemake software is available as a module in Puhti supercomputing environment. Also, a CSC user can easily install it in his/her own disk space (e.g., in /projappl directory) if a specific version of Snakemake is desired. The following toy example illustrates how a Snakemake workflow can be deployed at CSC.
+
+Please make sure that you have a [user account at CSC](../../accounts/how-to-create-new-user-account.md) and are a member of a project which [has access to the Puhti service](../../accounts/how-to-add-service-access-for-project.md) before start running workflows on Puhti. Please avoid launching heavy Snakemake workflow jobs on **login nodes** and use interactive or batch jobs instead. More information on using interactive jobs can be found on 
+[CSC documentation](../../computing/running/interactive-usage.md).
+
+Let's use the following toy Snakemake script, Snakefile (with a capital S and no file extension), for the illustration:
+
+```bash
+
+rule all:
+        input: "CAPITAL_CASE.txt"
+
+rule say_hello:
+        output: "smaller_case.txt"
+        shell:
+                """
+                echo "hello-world greetings from csc to snakemake community !" > smaller_case.txt
+                """
+rule capitalise:
+        input: "smaller_case.txt"
+        output: "CAPITAL_CASE.txt"
+        shell:
+                """
+                tr '[:lower:]' '[:upper:]' < {input} > {output}
+                """
+```
+As snakemake is installed as a module on Puhti, one can simply load it without needing to install it separately. Here is a bash script for running the above toy example:
+```bash
+module load snakemake/7.17.1
```

Comment:
General idea: should we consider upgrading to 8.X.X? We could then just used the newest version of docs when it comes to slurm/hyperqueue integration and avoid any misunderstandings.

Reply:
This is something I considered earlier. Plan is to install recent version of snakemake (as  non-containerised ) module and make it compatible with other container environments that users may come up with.  Yes. this needs bit of testing.

Reply:
I am leaving it up to your decision :) Feel free to close this thread.

Reply:
No worries. I pushed one more version with recent version of snakemake.